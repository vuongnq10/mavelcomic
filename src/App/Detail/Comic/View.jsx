import T from 'prop-types';
import React from 'react';
import Card from 'react-bootstrap/Card';

import Spinner from 'components/Spinner';
import styles from './styles.css';

const View = ({ loading, comic }) => (
  !loading ? <div className={styles.padding}>
    <Card>
      <Card.Header>{comic.title}</Card.Header>
      <Card.Body>
        <div className="row">
          <div className="col-lg-3 col-12">
            <img
              src={`${comic.thumbnail.path}.${comic.thumbnail.extension}`}
              alt={comic.title}
              className="img-thumbnail"
            />
          </div>
          <div className="col-lg-9 col-12">
            <Card.Title>Description</Card.Title>
            <Card.Text>{comic.description || 'N/A'}</Card.Text>
          </div>
        </div>
      </Card.Body>
    </Card>
  </div> : <Spinner />
);

View.propTypes = {
  loading: T.bool.isRequired,
  comic: T.shape({
    name: T.string,
  }).isRequired,
};

export default View;
