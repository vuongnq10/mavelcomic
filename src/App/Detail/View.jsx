import T from 'prop-types';
import React from 'react';

import Spinner from 'components/Spinner';

const View = props => (
  !props.loading ? <div className="container">
    {props.hero.name}
  </div> : <Spinner />
);

View.propTypes = {
  loading: T.bool.isRequired,
  hero: T.shape({
    name: T.string.isRequired,
  }).isRequired,
};

export default View;
