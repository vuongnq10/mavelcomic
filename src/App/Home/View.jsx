import T from 'prop-types';
import React from 'react';

import Spinner from 'components/Spinner';

import Chacrater from './Character';

const View = props => (
  <React.Fragment>
    <div className="container">
      <h1>List of heroes you will love</h1>
      {props.characters.map((item, i) => <Chacrater {...item} key={i} />)}
    </div>
    {props.loading && <Spinner />}
  </React.Fragment>
);

View.propTypes = {
  characters: T.arrayOf(T.object).isRequired,
  loading: T.bool.isRequired,
};

View.defaultProps = {
  characters: [],
};

export default View;
