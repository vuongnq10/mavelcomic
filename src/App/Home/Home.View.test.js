import React from 'react';
import renderer from 'react-test-renderer';

import View from './View';

it('Renders Home View component: loading=true', () => {
  const view = renderer
    .create(<View characters={[]} loading={true} />)
    .toJSON();

  expect(view).toMatchSnapshot();
});

it('Renders Home View component loading=false', () => {
  const view = renderer
    .create(<View characters={[]} loading={false} />)
    .toJSON();

  expect(view).toMatchSnapshot();
});