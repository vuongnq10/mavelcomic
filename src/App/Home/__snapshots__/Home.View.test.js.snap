// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home View component loading=false 1`] = `
<div
  className="container"
>
  <h1>
    List of heroes you will love
  </h1>
</div>
`;

exports[`Renders Home View component: loading=true 1`] = `
Array [
  <div
    className="container"
>
    <h1>
        List of heroes you will love
    </h1>
</div>,
  <div
    className={undefined}
>
    <div
        className={undefined}
    >
        <div />
        <div />
        <div />
        <div />
    </div>
</div>,
]
`;
